### ISSUE TITLE
Analysis of AutoQC results

### ISSUE THREAD

**Issue by s-good** [IQuOD/AutoQC#197]
The first 10,000 profiles in the QuOTA dataset have been processed through AutoQC. A quick look at the results reveals some interesting things. The method I followed was:

- Perform an 'AND' of the outcomes of all pairs of test (to see if some work best in combination with others).
- Exclude all tests that did not correctly flag any profiles.
- For a range of different thresholds, combine tests that have a false positive rate less than or equal to the threshold and see what the true positive rate was.

Initially the aim was to find a set of tests that have a very low false positive rate which can be assumed to contain bad data without further examination so only very low thresholds were considered. The ROC plot for this was:

![roc1](https://cloud.githubusercontent.com/assets/8092601/23836429/e607f616-076f-11e7-87ac-08feaf009f37.png)

Interesting points include:

True positive rate (%) | False positive rate (%)
--------------------------- | ----------------------------
4.24 | 0.00
56.30 | 0.13

The first is the highest TPR without rejecting a bad profile, the second is the point at which the FPR starts to increase rapidly. 56% seems a good amount of rejects. The combination of tests is a bit strange.

First case: ['Argo_spike_test', 'ICDC_aqc_07_spike_check', 'Argo_gradient_test&Argo_impossible_date_test', 'CSIRO_long_gradient&loose_location_at_sea', 'CoTeDe_Argo_density_inversion&loose_location_at_sea', 'CoTeDe_GTSPP_WOA_normbias&CoTeDe_GTSPP_spike_check', 'CoTeDe_GTSPP_profile_envelop&EN_std_lev_bkg_and_buddy_check', 'CoTeDe_Morello2014&loose_location_at_sea', 'CoTeDe_WOA_normbias&EN_constant_value_check', 'CoTeDe_gradient&EN_std_lev_bkg_and_buddy_check', 'EN_background_check&EN_increasing_depth_check']

Second case: ['Argo_global_range_check', 'Argo_spike_test', 'ICDC_aqc_07_spike_check', 'loose_location_at_sea', 'Argo_gradient_test&CoTeDe_GTSPP_profile_envelop', 'Argo_gradient_test&CoTeDe_WOA_normbias', 'Argo_gradient_test&EN_background_check', 'Argo_impossible_date_test&ICDC_aqc_09_local_climatology_check', 'Argo_regional_range_test&CoTeDe_WOA_normbias', 'CSIRO_long_gradient&CoTeDe_GTSPP_spike_check', 'CoTeDe_Argo_density_inversion&EN_constant_value_check', 'CoTeDe_GTSPP_WOA_normbias&EN_constant_value_check', 'CoTeDe_WOA_normbias&EN_increasing_depth_check', 'CoTeDe_WOA_normbias&EN_spike_and_step_suspect', 'EN_background_available_check&EN_spike_and_step_suspect', 'EN_background_check&EN_increasing_depth_check', 'EN_std_lev_bkg_and_buddy_check&ICDC_aqc_02_crude_range', 'ICDC_aqc_09_local_climatology_check&WOD_range_check']

The & is used for when two tests have been anded together. These seem to be very important, e.g. loose location at sea with others.

The profiles that were rejected in the second case were then put aside and the same procedure applied to the remainder but allowing for larger FPR. The ROC plot was:

![roc2](https://cloud.githubusercontent.com/assets/8092601/23836585/88784f5c-0772-11e7-8a7e-d91159c9c5d6.png)

There isn't a clear cut off to choose but about 60% TPR is achievable (i.e. overall about 85% TPR) for about 20% FPR. 

It will be interesting to see if the results hold for the full dataset and whether it is possible to use e.g. the machine learning approach to get a better TPR.

**Comment by s-good** [comment]
Update - using a slightly different methodology has given slightly better stats (new method is combine all tests with 0 FPR first then add in more tests one at a time in order of their ratio of TPR to FPR; I have also added in ANDed combinations of up to four tests rather than the up to two that was running before). I've found a combination with 56.71% TPR, 0.011% FPR (which I think may be just one profile incorrectly flagged).

It seems that there are quite a few profiles that are challenging to catch and some containing bad data that are never flagged at all. It would be interesting to find out the reason why these were flagged in QuOTA.

![roc](https://cloud.githubusercontent.com/assets/8092601/23871684/59837814-0823-11e7-9af6-0f7291ea1b4b.png)

**Comment by BillMills** [comment]
Eliminating over half of bad profiles with only a single mistake in 10k is a great start! After #195 and #196 are addressed, I'd like to investigate whether there are any 'hidden gems' in our collection of QC tests: tests that seem mediocre globally, but perform very well in one particular region, or when conducted with one particular probe type. Will throw this at some learning algorithms and see if anything shakes out.

**Comment by s-good** [comment]
This issue is old and superseded by newer results so I'm going to close it.
